File: Tawaf/app/src/main/java/me/mxcsyounes/tawaf/models/State.java
package me.mxcsyounes.tawaf.models;

public class State {

    public void setStateHajar(int stateHajar) {
        this.stateHajar = stateHajar;
    }

    public void setStateRokn(int stateRokn) {
        this.stateRokn = stateRokn;
    }

    private int stateHajar;

    public State() {
    }

    private int stateRokn;

    public State(int stateHajar, int stateRokn) {
        this.stateHajar = stateHajar;
        this.stateRokn = stateRokn;
    }

    public int getState() {
        if (stateHajar == 1) {
            if (stateRokn == 1) return 1;
            else return 2;
        } else {
            if (stateRokn == 1) return 4;
            else return 3;
        }
    }
}

File: Tawaf/app/src/main/java/me/mxcsyounes/tawaf/models/State.kt
package me.mxcsyounes.tawaf.models;

data class State(var stateHajar: Int, var stateRokn: Int) {

    //no argument constructor
    constructor() : this(0, 0)

    //function to get state in the haram
    fun getState() =
            if (stateHajar == 1) {
                if (stateRokn == 1) 1
                else 2
            } else {
                if (stateRokn == 1) 4
                else 3
            }

}
